bindings/kafka/kafka.go

```diff
@@ -110,17 +121,19 @@ func (b *Binding) Read(ctx context.Context, handler bindings.Handler) error {
 		}
 
 		// If the component's context has been canceled, do not re-subscribe
-		if b.subscribeCtx.Err() != nil {
+		select {
+		case <-ctx.Done():
 			return
+		default:
 		}
 
-		err := b.kafka.Subscribe(b.subscribeCtx)
+		err := b.kafka.Subscribe(ctx)
```

Comment:
I see. What is the point of re-subscribing after the read context has been cancelled?

Reply:
I believe (please correct me if I'm wrong) that Kafka maintains a single, persistent connection that's started in Init and is used to both publish and subscribe (to all topics). If you tie the lifecycle of that connection to a subscription, cancling the subscription means the connection would be closed and so you'd not be able to publish anymore (and other subscriptions added to the same connection would be dropped)

Reply:
Thanks @ItalyPaleAle, I guess I didn't totally appreciate the fact that Subscribe restarts the whole kafka client with the new topic. Even so, does it matter? As I understand, calling `Read` multiple times with different handlers isn't supported anyway since we are overriding the existing handlers for the defined topics:

```go
	for _, t := range b.topics {
		b.kafka.AddTopicHandler(t, handlerConfig)
	}
```

My feeling is  we should remove this second `Subscribe` completely. 

The pubsub will be using a different client so we don't need to worry about that one.


I'll switch the context to a `Background`.

Reply:
I did not write that code so I am not sure, but I am happy to accept your proposed change if you can please validate it (essentially try to add and remove topics dynamically and see if that causes any issues)... I think we have a conformance test that does that, but may need some extra work on top.